pkg/crc/image/image.go

```diff
@@ -0,0 +1,113 @@
+package image
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"runtime"
+	"strings"
+
+	crcpreset "github.com/code-ready/crc/pkg/crc/preset"
+	"github.com/containers/image/v5/copy"
+	"github.com/containers/image/v5/signature"
+	"github.com/containers/image/v5/transports/alltransports"
+	"github.com/containers/image/v5/types"
+	v1 "github.com/opencontainers/image-spec/specs-go/v1"
+)
+
+var (
+	registryURI = "quay.io"
+	orgName     = "crcont"
+)
+
+type OCIImageHandler struct {
+	ImageName string
+	ImageTag  string
+	Progress  io.Writer
+}
+
+func (img *OCIImageHandler) uri() string {
+	return strings.Join([]string{registryURI, orgName, img.ImageName}, "/")
+}
+
+func (img *OCIImageHandler) parse(image string) (types.ImageReference, error) {
+	return alltransports.ParseImageName(image)
+}
+
+func (img *OCIImageHandler) policyContext() (*signature.PolicyContext, error) {
+	policy := &signature.Policy{Default: []signature.PolicyRequirement{signature.NewPRInsecureAcceptAnything()}}
+	policyContext, err := signature.NewPolicyContext(policy)
+	if err != nil {
+		return nil, fmt.Errorf("error creating security context: %s", err.Error())
+	}
+
+	return policyContext, nil
+}
+
+func (img *OCIImageHandler) getSystemContext() *types.SystemContext {
+	return &types.SystemContext{
+		OSChoice:           "linux",
+		ArchitectureChoice: runtime.GOOS,
```

Comment:
not `GOARCH`?

Reply:
Arch is taken as default of running host so we don't need to explicit mention it. `OSChoice` we need because all the images are linux based.

Reply:
But this is `ArchitectureChoice` that you are setting, not `OSChoice`, `ArchitectureChoice: runtime.GOARCH` makes sense to me, so does `OSChoice: runtime.GOOS`. Here we have `ArchitectureChoice: runtime.GOOS` which looks unexpected?

Reply:
@cfergeau yes I remove that `ArchitectureChoice: runtime.GOOS` also I can't set `OSChoice: runtime.GOOS` because our images are only linux based not `darwin` or `window` based which cause issue during the image pull.
```
choosing an image from manifest list docker://quay.io/crcont/podman-vfkit-bundle:2.4.1: no image found in image index for architecture arm64, variant "", OS darwin
```